### ISSUE TITLE
Using with Sanic

### ISSUE THREAD

**Issue by ysfchn** [r0x0r/pywebview#1074]
### Specification
- pywebview version: 4.0.2
- operating system: GNU/Linux, Pop!_OS 22.04 LTS, x86_64, Kernel 6.2.0-76060200-generic
- web renderer: gtk

### Description

I'm trying to use Sanic as server (means without Bottle involved at all, because I'm planning to do browser-server communication on my own.). I know pywebview supports WSGI applications, but I feel like it would be better to start pywebview from Sanic, instead of letting pywebview to boot Sanic. So it won't mix up/conflict with other processes/workers of Sanic, and I can create another child processes under Sanic when I want.

Here is a partially-working snippet after playing with parameters with Sanic listeners, (works for remote URLs - like `https://example.com`, but not for local hosts, not sure why, maybe there is wrong on my side) 
  
```py
from sanic import Sanic, html
import webview

app = Sanic("my_app")

def get_serve_url(app : Sanic):
    return f"http://{app.state.host}:{app.state.port}"

async def start_window(app : Sanic):
    webview.create_window("Hello world", get_serve_url(app), confirm_close = True, 
        server = None, width = 1300, height = 750
    )
    webview.start(debug = True, server = None)
    app.stop()

@app.get("/")
async def home(request):
    return html("""
    <!DOCTYPE html>
    <html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>Document</title>
    </head>
    <body>
        <p>Test!</p>
    </body>
    </html>
    """)

if __name__ == "__main__":
    app.add_task(start_window)
    app.run(single_process = True)
```

Thanks!
  
### Practicalities

- NO I am willing to work on this issue myself. Well, I would like to help anyway, but not sure if I can contribute via PR.

- NO I am prepared to support this issue financially.

**Comment by ysfchn** [comment]
After experimenting more, I found this solution. It also handles processes like if webview is closed before server, server will be killed, and vice versa.

```py
from sanic import Sanic, text
import webview
import os
import signal

# Dummy server with no functionality.
class NoServer(webview.http.BottleServer):
    @classmethod
    def start_server(cls, urls, http_port):
        self.common_path = "."
        server = cls()
        server.running = True
        server.address = ""
        return server.address, ".", server

# Create Sanic app.
app = Sanic("my_app")

def get_serve_url(app : Sanic):
    return f"http://{app.state.host}:{app.state.port}/"

def start_window(url : str):
    webview.create_window("Hello world", url, confirm_close = True, width = 1300, height = 750)
    try:
        webview.start(server = NoServer)
    except KeyboardInterrupt: 
        pass
    # If webview window closes before the server, stop the server.
    os.kill(os.getppid(), signal.SIGINT)

@app.get("/")
async def home(request):
    return text("Hello world")

@app.main_process_ready
async def server_start(app : Sanic, loop):
    # Start webview along with server.
    app.manager.manage("pywebview", start_window, {"url": get_serve_url(app)})

if __name__ == "__main__":
    app.run()
```

**Comment by SHI1992** [comment]
Hello, I tried running your code with Sanic + pywebview. It works normally, and the desktop application can start and close properly without any errors.

However, I noticed that after closing the desktop application, two Python processes are not killed.

I set Sanic's workers = 1 with app.run(host="127.0.0.1", port=8000, workers=1). When the application starts, there are a total of 5 Python processes. After closing the application, 3 processes are automatically terminated, but 2 Python processes remain hanging.

![Image](https://github.com/user-attachments/assets/cc0cb868-5332-48c6-accb-17119e1005ae)
![Image](https://github.com/user-attachments/assets/98826852-e728-499e-9481-ceedd09c4326)

As a result, repeatedly starting and closing the application leads to more and more Python processes being left running.

How can I resolve this issue?

Thank you.
